tests/zim/test_zim_creator.py

```diff
@@ -540,6 +542,82 @@ def test_check_metadata(tmp_path):
         Creator(tmp_path, "").config_dev_metadata(LongDescription="T" * 5000).start()
 
 
+@pytest.mark.parametrize(
+    "tags",
+    [
+        (
+            "wikipedia;_category:wikipedia;_pictures:no;_videos:no;_details:yes;"
+            "_ftindex:yes"
+        ),
+        (
+            [
+                "wikipedia",
+                "_category:wikipedia",
+                "_pictures:no",
+                "_videos:no",
+                "_details:yes",
+                "_ftindex:yes",
+            ]
+        ),
+    ],
+)
+@patch("zimscraperlib.zim.creator.logger", autospec=True)
+def test_start_logs_metadata_log_contents(mocked_logger, png_image, tags, tmp_path):
+    mocked_logger.isEnabledFor.side_effect = lambda level: level == logging.DEBUG
+    fpath = tmp_path / "test_config.zim"
+    with open(png_image, "rb") as fh:
+        png_data = fh.read()
+    # Using `with creator:` would call start() twice, complicating
+    # the assert below.
+    creator = Creator(fpath, "").config_metadata(
+        Name="wikipedia_fr_football",
+        Title="English Wikipedia",
+        Creator="English speaking Wikipedia contributors",
+        Publisher="Wikipedia user Foobar",
+        Date="2009-11-21",
+        Description="All articles (without images) from the english Wikipedia",
+        LongDescription="This ZIM file contains all articles (without images)"
+        " from the english Wikipedia by 2009-11-10. The topics are...",
+        Language="eng",
+        License="CC-BY",
+        Tags=tags,
+        Flavour="nopic",
+        Source="https://en.wikipedia.org/",
+        Scraper="mwoffliner 1.2.3",
+        Illustration_48x48_at_1=png_data,
+        TestMetadata="Test Metadata",
+    )
+    creator.start()
+    creator.finish()
+    mocked_logger.debug.assert_has_calls(
+        [
+            call("Metadata: Creator = English speaking Wikipedia contributors"),
+            call("Metadata: Date = 2009-11-21"),
+            call(
+                "Metadata: Description = All articles (without images) from the "
+                "english Wikipedia"
+            ),
+            call("Metadata: Flavour = nopic"),
+            call("Metadata: Illustration_48x48@1 MD = PNG"),
+            call("Metadata: Language = eng"),
+            call("Metadata: License = CC-BY"),
+            call(
+                "Metadata: LongDescription = This ZIM file contains all articles "
+                "(without images) from the english Wikipedia by 2009-11-10. "
+                "The topics are..."
+            ),
+            call("Metadata: Name = wikipedia_fr_football"),
+            call("Metadata: Publisher = Wikipedia user Foobar"),
+            call("Metadata: Relation = None"),
+            call("Metadata: Scraper = mwoffliner 1.2.3"),
+            call("Metadata: Source = https://en.wikipedia.org/"),
+            call(f"Metadata: Tags = {tags}"),
```

Comment:
There is an implicit conversion here from array to string with values separated by commas, isn't it? I would rather always assert for the expected value in all cases.

Reply:
That'd be nice, but at least tags is a parameterized constant.

When trying to replace {tags} with a constant, I got two exceptions, one interupting the other, that expected different formats for tags, one a [foo, bar] list and one a semicolon-separated list (like foo;bar); satisfying either would make the other fire first.

Using {tags} seems to work for reasons I do not understand. I could also handle Tags separately checking for either format, but that seems even worse.

Reply:
OK ; this means we let Python decide how to transform this into a string, but it is probably the proper solution since it is likely that the same will have in Creator itself.